spring-beans/src/main/java/org/springframework/beans/factory/support/AbstractBeanFactory.java

```diff
@@ -583,6 +584,27 @@ else if (typeToMatch.hasGenerics() && containsBeanDefinition(beanName)) {
 				// Generics potentially only match on the target class, not on the proxy...
 				RootBeanDefinition mbd = getMergedLocalBeanDefinition(beanName);
 				Class<?> targetType = mbd.getTargetType();
+
+				String scope = mbd.getScope();
+				if (targetType == null && scope != null && !scope.isEmpty()) {
+					String targetBeanName = "scopedTarget." + beanName;
+					if (containsBeanDefinition(targetBeanName)) {
+						RootBeanDefinition targetMbd = getMergedLocalBeanDefinition(targetBeanName);
+
+						ResolvableType targetResolvableType = targetMbd.targetType;
+						if (targetResolvableType == null) {
+							targetResolvableType = targetMbd.factoryMethodReturnType;
+							if (targetResolvableType == null) {
+								targetResolvableType = ResolvableType.forClass(targetMbd.getBeanClass());
+							}
+						}
+
+						if (typeToMatch.isAssignableFrom(targetResolvableType)) {
```

Comment:
considering 

https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/ObjectUtils.html#defaultIfNull(T,T)

https://github.com/spring-projects/spring-framework/pull/34587

we should be able to reduce all redundancy by using the internal ternary operator hidden inside `ObjectUtils.defaultIfNull`

```suggestion
						if (typeToMatch.isAssignableFrom(ObjectUtils.getIfNull(targetResolvableType ,ObjectUtils.getIfNull(targetMbd.factoryMethodReturnType, ResolvableType.forClass(targetMbd.getBeanClass()))))) {
```

Reply:
but of course this can be done afterward or not at all, as its working an acceptable anyways.

Reply:
Thank you for the suggestion and for pointing me to the ObjectUtils.defaultIfNull from Apache Commons Lang!

I initially wasn't able to use any defaultIfNull method because I was working with org.springframework.util.ObjectUtils, which doesn't have this method in our current version.

Your approach using nested ternary operators looks more concise and elegant. I agree this would be a great improvement, but I think we should leave it for a separate refactoring PR in the future. This way, we can keep the current PR focused on fixing the original issue with clean, minimal changes.

Thanks again for your approval and the valuable feedback throughout this PR!

Reply:
I would suggest merging it as it is and incorporating the refactoring into the Util PR (https://github.com/spring-projects/spring-framework/pull/34587) so we don’t add empty code and can deliver a practical example right away. 

If it’s merged before, then you can, of course, try it here. I would not suggest mixing the PRs now.

Thanks for your heartwarming feedback—teamwork makes the dream work!